main_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```

Comment:
This is original work, so @kashifest we should switch to Metal3 Authors right?

Reply:
Yes